PyDSS/utils/simulation_utils.py

```diff
@@ -27,7 +27,7 @@ def append(self, val):
         self._buf.append(val)
 
     def average(self):
-        if len(self._buf) < self._window_size:
```

Comment:
Can you figure out why your window size is 0? I suspect the best fix is add this to line 21.
```
if window_size < 1:
    raise ValueError(f"window_size={window_size} is invalid")
```

Reply:
I have `simulation_type = "snapshot"` in `[project]`. The `window_size` gets set to zero here:
https://github.com/NREL/PyDSS/blob/8a208ea5aa9d64919d60ce06854c14f1fa64eee7/PyDSS/metrics.py#L652-L654
because the numerator is less than 1/2 of the denominator. I thought that these input values would not matter because I am running a single time step. Are the averages meant to be time averages and therefor the `average` function should return the single value in `self._buf`? Or can the `average` function be used for spatial averages as well?

Reply:
This report will create the metrics listed in this file: https://github.com/NREL/PyDSS/blob/8a208ea5aa9d64919d60ce06854c14f1fa64eee7/PyDSS/node_voltage_metrics.py#L100.

A moving average of voltage violations is meaningless in this case, so we should be able to find a way to skip it. For the time being you could adjust the settings so that it goes through. Or we could merge your PR as is. My concern is that this class by itself is nonsensical if window_size is 0.

Reply:
> A moving average of voltage violations is meaningless in this case, so we should be able to find a way to skip it.

Agreed. I think that my change makes them `null` right? I still would like to see the max/min voltage values reported. (As an aside, I'm working towards running a stochastic hosting capacity analysis so I want to record the max/min voltage for 1,000s of scenarios).


> My concern is that this class by itself is nonsensical if window_size is 0.

Should I add ` or self._window_size == 0` to the `average` method?

Reply:
How about we set the minimum window_size to 1?
https://github.com/NREL/PyDSS/blob/master/PyDSS/metrics.py#L652

One of my teams conducted large-scale hosting capacity analysis using this tool and others. Please contact me directly if you'd like to discuss our methodology. We'd be happy to share what we learned.

Reply:
Do you want me to revert https://github.com/NREL/PyDSS/pull/113/commits/be18043297c170817cb34db29027e944460e1de3 ? I.e. do you want `np.Nan` when there are zero samples?

I sent you a message on LinkedIn (NREL's email filters don't let much through).

Reply:
Yeah, please revert if you don't mind. I'll come back around eventually and make the CircularBuffer class throw an error if the window size is 0/nonsensical. Thanks for the fix.